lib/sidekiq/scheduler.rb

```diff
@@ -399,6 +404,24 @@ def set_schedule_state(name, state)
         Sidekiq.redis { |r| r.hset(schedules_state_key, name, JSON.generate(state)) }
       end
 
+      # Adds a Hash with schedule metadata as the last argument to call the worker.
+      # @example when parameter is an array
+      #   arguments_with_metadata(['arg1']) #=> ['arg1', {:scheduled_at => 1486039908.574729}]
+      # @example when parameter is a hash
+      #   arguments_with_metadata({arg1: 'value'}) #=> {:arg1 => 'value', :metadata => {:scheduled_at => 1486039908.574729}}
+      #
+      # @param args [Array|Hash]
+      # @return [Array|Hash] arguments with added metadata
+      def arguments_with_metadata(args)
+        metadata_hash = {scheduled_at: Time.now.to_f}
```

Comment:
See https://github.com/jmettraux/rufus-scheduler#last_time. Figure out if one of [`#last_time`, `#previous_time`, `#scheduled_at`] returns the timestamp that hit the rufus-scheduler trigger